NumLib/Fem/ShapeMatrixPolicy.h

```diff
@@ -107,8 +107,11 @@ struct EigenDynamicShapeMatrixPolicy
 
 /// Default choice of the ShapeMatrixPolicy.
 template <typename ShapeFunction, unsigned GlobalDim>
+#if EIGEN_DYNAMIC_MATRICES
```

Comment:
Could you pull the `template<...>` line inside the `#ifdef', please. I think, that's more clear.